include/circt/Dialect/FIRRTL/FIRRTLCanonicalization.td

```diff
@@ -125,34 +127,35 @@ def GetWidthAsIntAttr : NativeCodeCall<
 
 // node(x,name),!dnt -> node(x, "")
 def DropNameNode : Pat<
-  (NodeOp $x, $name, $annotations, $inner_sym),
-  (NodeOp $x, (GetEmptyString), $annotations, $inner_sym),
+  (NodeOp $x, $name, $annotations, $inner_sym, $nameKind),
+  (NodeOp $x, (GetEmptyString),  $annotations, $inner_sym, (GetDroppableNameAttr)),
   [(uselessNameAttr $name), (NonEmptyAttr $name), (NullAttr $inner_sym)]>;
```

Comment:
rename uselessNameAttr?

Reply:
A also thought the DropName* weren't being used any more.  They probably shouldn't.  We shouldn't drop names anymore.  A droppable name should be kept and moved to expressions.

Reply:
`uselessNameAttr` is intentionally used because we want to drop names with `_` prefix. I agree that these canonicalizers are not used anymore so I'll do in a follow-up.